server/sonar-server/src/main/java/org/sonar/server/webhook/ws/SearchAction.java

```diff
@@ -0,0 +1,136 @@
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.ORGANIZATION_KEY_PARAM;
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.PROJECT_KEY_PARAM;
+import static org.sonar.server.webhook.ws.WebhooksWsParameters.SEARCH_ACTION;
+import static org.sonar.server.ws.KeyExamples.KEY_ORG_EXAMPLE_001;
+import static org.sonar.server.ws.KeyExamples.KEY_PROJECT_EXAMPLE_001;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+import static org.sonarqube.ws.Webhooks.SearchWsResponse.newBuilder;
+
+public class SearchAction implements WebhooksWsAction {
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final SettingsFinder settingsFinder;
+
+  public SearchAction(DbClient dbClient, UserSession userSession, SettingsFinder settingsFinder) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.settingsFinder = settingsFinder;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+
+    WebService.NewAction action = controller.createAction(SEARCH_ACTION)
+      .setDescription("Search for webhooks associated to an organization or a project.<br/>")
+      .setSince("7.1")
+      .setResponseExample(Resources.getResource(this.getClass(), "example-webhooks-search.json"))
+      .setHandler(this);
+
+    action.createParam(ORGANIZATION_KEY_PARAM)
+      .setDescription("Organization key. If no organization is provided, the default organization is used.")
+      .setInternal(true)
+      .setRequired(false)
+      .setExampleValue(KEY_ORG_EXAMPLE_001);
+
+    action.createParam(PROJECT_KEY_PARAM)
+      .setDescription("Project key")
+      .setRequired(false)
+      .setExampleValue(KEY_PROJECT_EXAMPLE_001);
+
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+
+    String project = request.param(PROJECT_KEY_PARAM);
+
+    userSession.checkLoggedIn();
+
+    List<SearchElement> elements = doHandle(project);
+
+    writeResponse(request, response, elements);
+
+  }
+
+  private List<SearchElement> doHandle(String project) {
+
+    try (DbSession dbSession = dbClient.openSession(true)) {
+
+      List<Setting> settings = new ArrayList<>();
+
+      if (isNotBlank(project)) {
+        Optional<ComponentDto> component = dbClient.componentDao().selectByKey(dbSession, project);
+        if (component.isPresent()) {
+          userSession.checkComponentPermission(ADMIN, component.get());
+          settings = settingsFinder.loadProjectSettings(dbSession, component.get(), "sonar.webhooks.project");
+        }
+      } else {
+        userSession.checkIsSystemAdministrator();
+        settings = settingsFinder.loadGlobalSettings(dbSession, ImmutableSet.of("sonar.webhooks.global"));
+      }
+
+      return settings
+        .stream()
+        .map(Setting::getPropertySets)
+        .flatMap(Collection::stream)
+        .map(map -> new SearchElement("", map.get("name"), map.get("url")))
```

Comment:
empty key ? I suppose that it will come soon with new table `webhooks` ?

Reply:
indeed.